web/packages/teleport/src/Discover/Shared/usePoll.ts

```diff
@@ -2,20 +2,17 @@ import { useEffect, useRef, useState } from 'react';
 
 export function usePoll<T>(
   callback: (signal: AbortSignal) => Promise<T | null>,
-  timeout: number,
   enabled: boolean,
   interval = 1000
-) {
+): T | null {
```

Comment:
Are you noticing a bug here wherein Typescript doesn't seem to pick up that this can return `null`? I notice that my editor (vscode) doesn't seem to pick it up, and if I take out `&& result` from this line

https://github.com/gravitational/teleport/blob/b20144dd090fd4128502c1cf2ac978f7de87d25f/web/packages/teleport/src/Discover/Desktop/DiscoverDesktops/DiscoverDesktops.tsx#L109

a run of `yarn type-check` doesn't complain at me

```
$ yarn type-check
yarn run v1.22.19
$ tsc --noEmit
✨  Done in 43.69s.
```

Reply:
Yeah I noticed this as well, it just returns `T`. I was just trying to be strict with typings by adding this 🤷

Reply:
Huh, seems like a ts bug. It might be worth adding explicit typings to places where the automatic typing is failing us and see if that helps.